### PR TITLE
Store VS assignments on the DB

### PR DESCRIPTION
**Pull request by marcospri**
For https://github.com/hypothesis/lms/issues/3329

Handles VitalSource assignments like any other type of assignment on Canvas.

Testing as https://github.com/hypothesis/lms/pull/3126 but using VS assignments as well.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ export default function BasicLTILaunchApp() {
     }
 
     // Don't report a submission until the URL has been successfully fetched.
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js:237]: Not sure how to rephrase this comment "... until the content has been configured"?

File: lms/views/basic_lti_launch.py [hunk 2]
```diff
@@ -122,26 +122,6 @@ def legacy_canvas_file_basic_lti_launch(self):
...
-        Hypothesis's LMS app generates the form parameters needed to load
-        VitalSource's book viewer using an LTI launch. The LMS frontend then
-        renders these parameters into a form and auto-submits the form to perform
-        an authenticated launch of the VS book viewer.
-        """
-        self.sync_lti_data_to_h()
-        self.store_lti_data()
-        self.context.js_config.maybe_enable_grading()
-        self.context.js_config.add_vitalsource_launch_config(
-            self.request.params["book_id"], self.request.params.get("cfi")
```
Comment (marcospri) [inline lms/views/basic_lti_launch.py:141]: We don't need to send any params to SpeedGrader as we can read the information from the DB assignment.

File: lms/views/predicates/_lti_launch.py [hunk 3]
```diff
@@ -186,13 +186,6 @@ def __call__(self, context, request):
         return ("canvas_file" in request.params) == self.value
 
 
-class VitalSourceBook(Base):
```
Comment (marcospri) [inline lms/views/predicates/_lti_launch.py:189]: VitalSource assignments become regular DB-Configured ones.

**Review by seanh — APPROVED** [review]

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js (hunk 1 above)
Comment (robertknight) [reply lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js:237]: I suggest something like "Don't report a submission until we have the data needed to display the assignment content".

**Pull request merged**